src/live.rs

```diff
@@ -160,6 +165,197 @@ pub fn pxe_ignition_unwrap(config: &PxeIgnitionUnwrapConfig) -> Result<()> {
+        &config.delete,
+    )?;
+    embed.write_kargs(&new_kargs)?;
+    holder.complete();
+    Ok(())
+}
+
+pub fn iso_kargs_reset(config: &IsoKargsResetConfig) -> Result<()> {
+    let mut holder = CopiedFileHolder::new(&config.input, config.output.as_ref())?;
+    let mut embed = KargEmbedAreas::for_file(&mut holder.file)?;
+    let default_kargs = embed.get_default_kargs()?;
+    embed.write_kargs(&default_kargs)?;
+    holder.complete();
+    Ok(())
+}
+
+pub fn iso_kargs_show(config: &IsoKargsShowConfig) -> Result<()> {
+    let mut file = OpenOptions::new()
+        .read(true)
+        .open(&config.input)
+        .chain_err(|| format!("opening {}", &config.input))?;
+    let mut embed = KargEmbedAreas::for_file(&mut file)?;
+    let kargs = if config.default {
+        embed.get_default_kargs()?
+    } else {
+        embed.get_current_kargs()?
+    };
+    println!("{}", kargs);
+    Ok(())
+}
+
+struct KargEmbedAreas<'a> {
+    file: &'a mut File,
+    length: usize,
+    default_kargs_offset: u64,
+    kargs_offsets: Vec<u64>,
+}
+
+impl<'a> KargEmbedAreas<'a> {
+    fn for_file(file: &'a mut File) -> Result<Self> {
+        let mut buf: [u8; 8] = [0; 8];
+        // The ISO 9660 System Area is 32 KiB. Karg embed area information is located in the 72 bytes
+        // before the initrd embed area (see EmbedArea below):
+        // 8 bytes: magic string "corekarg"
+        // 8 bytes little-endian: length of karg embed areas
+        // 8 bytes little-endian: offset to default kargs
+        // 8 bytes little-endian x 6: offsets to karg embed areas
+        file.seek(SeekFrom::Start(
+            32768 - COREOS_IGNITION_HEADER_SIZE - COREOS_KARG_EMBED_AREA_HEADER_SIZE,
+        ))
+        .chain_err(|| "seeking to karg embed header")?;
+        // magic number
+        file.read_exact(&mut buf)
+            .chain_err(|| "reading karg embed header")?;
+        if buf != COREOS_KARG_EMBED_AREA_HEADER_MAGIC {
+            bail!("No karg embed areas found; old or corrupted CoreOS ISO image.");
+        }
+        // length
+        file.read_exact(&mut buf)
+            .chain_err(|| "reading karg embed header")?;
+        let length: usize = u64::from_le_bytes(buf)
+            .try_into()
+            .chain_err(|| "karg embed area length too large to allocate")?;
+        // sanity-check against a reasonable limit
+        if length > COREOS_KARG_EMBED_AREA_MAX_SIZE {
+            bail!(
+                "karg embed area length larger than {} (found {})",
+                COREOS_KARG_EMBED_AREA_MAX_SIZE,
+                length
+            );
+        }
+
+        let metadata = file.metadata().chain_err(|| "reading metadata for ISO")?;
+        let iso_size = metadata.len();
+
+        // default kargs
+        file.read_exact(&mut buf)
+            .chain_err(|| "reading karg embed header")?;
+        let default_kargs_offset: u64 = u64::from_le_bytes(buf);
+        if default_kargs_offset > iso_size {
```

Comment:
We should check that the entire buffer is inside the ISO, so `if default_kargs_offset + length > iso_size`.